### PR TITLE
py-configspace: add main, 1.0.0, 1.0.1, 1.1.1, 1.1.2, 1.1.3, 1.1.4, fix url and fix cython dependency

### PR DESCRIPTION
**Pull request by mdorier**
This PR restricts the dependency on py-cython up to 0.29.36 for the py-configspace package for version < 1.0.0.
py-configspace just doesn't work with cython 3+, things fail with exceptions related to type conversions in pyx files.

This PR also adds the "main" version, directly pointing to git, as well as version 1.0.0 and 1.0.1, which don't depend on py-cython and have a slightly different URL (without capital letters), hence the url_for_version.

### CODE REVIEW

**Comment by mdorier** [comment]
@Kerilk could you review this PR?

**Review by Kerilk — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-configspace/package.py
Comment (Kerilk) [inline var/spack/repos/builtin/packages/py-configspace/package.py]:
```suggestion
    version("1.1.3", sha256="8b77e77bd1c286a57e35da87552e33052f6793ddbcc696a9fc62425f60739ac2")
    version("1.1.2", sha256="8cd77438f976ce65ce2d056fbd659d12ca1425fe230b737942261879b7c542f0")
```

**Comment by mdorier** [comment]
I added version 1.1.2 and 1.1.3, and I added myself as a maintainer. @Kerilk can you re-approve once the tests pass?

**Review by Kerilk — APPROVED** [review]
LGTM.

**Comment by mdorier** [comment]
I'll continue adding versions as they are released, until someone approves/merges this PR :-)

**Review by wdconinc — APPROVED** [review]
Thanks and apologies for the late merge.

**Pull request merged**
